scripts/build_llvm_compiler_rt.sh

```diff
@@ -0,0 +1,216 @@
+    validate_llvm_install_path "$LINUX_INSTALL_PATH" "Linux"
+
+    # Ask for source path
+    echo_step  "Source Path Setup"
+    echo
+    read -p "Enter the directory path where LLVM source directory (riscv-llvm) is located [default is $(pwd)]: " SOURCE_DIR
+    SOURCE_DIR=${SOURCE_DIR:-$(pwd)}
+
+    # Check if the provided SOURCE_DIR exists
+    if [[ -d "$SOURCE_DIR" ]]; then
+        # The directory exists; now check if it contains the riscv-llvm directory
+        if [[ ! -d "$SOURCE_DIR/riscv-llvm" ]]; then
+            echo "The directory does not contain 'riscv-llvm'. LLVM source will be cloned."
+            CLONE_LLVM_SOURCE=true
+        else
+            echo "Found 'riscv-llvm' directory in the specified path."
+        fi
+    else
+        # The directory does not exist, indicate that LLVM source needs to be cloned
+        echo "The specified directory does not exist. It will be created, and LLVM source will be cloned into it."
+        CLONE_LLVM_SOURCE=true
+        mkdir -p "$SOURCE_DIR" || { pretty_error "Failed to create the directory at $SOURCE_DIR."; exit 1; }
+    fi
+
+    # Confirm start of LLVM build process
+    echo_step "LLVM Compiler RT Build Process"
+    echo
+    read -p "The LLVM Compiler RT build process is ready to be started. Do you wish to start the process? [Y/n]: " start_setup
+    if [[ ! "$start_setup" =~ ^([yY][eE][sS]|[yY])$ ]] && [ -n "$start_setup" ]; then
+        pretty_error "User aborted the LLVM Compiler RT build process."
+        exit 1
+    fi
+}
+
+#--------------------------------------------------------------------------------------------------------------------------
+
+# Function to clone the required repositories
+clone_repositories() {
+    echo_step "Cloning repositories"
+
+    cd "$SOURCE_DIR" || { echo "Failed to change directory to SOURCE_DIR."; exit 1; }
+
+    if [ -d "riscv-llvm" ]; then
+        echo "LLVM project directory already exists. Skipping cloning."
+    else
+        echo "Cloning LLVM project..."
+        git clone https://github.com/llvm/llvm-project.git riscv-llvm || { echo "Failed to clone LLVM project."; exit 1; }
+        cd riscv-llvm || { echo "Failed to change directory to riscv-llvm."; exit 1; }
+        git checkout $LLVM_PROJECT_COMMIT_SHA || { echo "Failed to checkout specified commit for LLVM project."; exit 1; }
+        cd "$SOURCE_DIR" || { echo "Failed to return to SOURCE_DIR."; exit 1; }
+    fi
+
+    # Create the symbolic link and fail if the operation does not succeed
+    if [ ! -d "riscv-llvm/llvm/tools/clang" ]; then
+        cd riscv-llvm || { echo "Failed to enter riscv-llvm directory."; exit 1; }
+        ln -s ../../clang llvm/tools || { echo "Failed to create symbolic link for clang."; exit 1; }
+    else
+        echo "Symbolic link for clang already exists."
+    fi
+}
+
+# Function to compile LLVM Compiler RT for Baremetal
+compile_llvm_compiler_rt_baremetal() {
+    echo_step "Compiling LLVM Compiler RT for Baremetal"
+
```

Comment:
I do not see a check/question for overwriting an existing rt lib. Is it here? 

If there is an existing lib, ask the user to over write or not. If not abort.